scripts/llext_link_helper.py

```diff
@@ -32,56 +33,107 @@ def parse_args():
 def main():
 	parse_args()
 
+	TYPE_NONE = 0
+	TYPE_DATA = 1
+	TYPE_BSS = 2
+	last_type = TYPE_NONE
+
+	SHF_ALLOC = 2
+	SHF_EXECINSTR = 4
+
 	elf = ELFFile(open(args.file, 'rb'))
 
 	text_addr = int(args.text_addr, 0)
-	p = re.compile(r'(^lib|\.so$)')
-	module = p.sub('', args.file)
 
-	if elf.num_segments() != 0:
-		# A shared object type image, it contains segments
-		sections = ['.text', '.rodata', '.data', '.bss']
-		alignment = [0x1000, 0x1000, 0x10, 0x1000]
+	# File names differ when building shared or relocatable objects
```

Comment:
Does this really belong to this script? As opposed to its caller.

Reply:
we do deal with both cases internally, and passing a module name additionally to the file name would be redundant